kimchi/src/circuits/polynomials/xor.rs

```diff
@@ -501,32 +471,48 @@ fn init_xor<F: PrimeField>(
     )
 }
 
-/// Extends the xor rows to the full witness
+/// Extends the Xor rows to the full witness
+/// Panics if the words are larger than the desired bits
 pub fn extend_xor_rows<F: PrimeField>(
     witness: &mut [Vec<F>; COLUMNS],
     bits: usize,
     words: (F, F, F),
 ) {
-    let xor_witness: [Vec<F>; COLUMNS] = array::from_fn(|_| vec![F::zero(); num_xors(bits) + 1]);
+    let input1_big = words.0.to_biguint();
+    let input2_big = words.1.to_biguint();
+    let output_big = words.2.to_biguint();
+    if bits < big_bits(&input1_big) || bits < big_bits(&input2_big) || bits < big_bits(&output_big)
```

Comment:
I think you should definitely implement `big_bits` as a `BigUnitHelpers` trait instead, so the code here will look like

```rust
if bits < input1_big.bit_len() || bits < input2_big.bit_len() || bits < output_big.bit_len()
```

Reply:
done!